### PR TITLE
Clear window.location.hash when drawing

### LINKED ISSUE #6: window.location.hash doesn't reset when drawing
after saving a mathpaste, the URL in address bar becomes something like `https://akuli.github.io/mathpaste/#saved:-Lxk8niJxIgyqqaKfZal`

changing something in the editor will change the url to `https://akuli.github.io/mathpaste/#` to let the user know that what they are looking at no longer matches the saved paste

but doing something with the drawing area doesn't change url

### PR DESCRIPTION
**Pull request by PurpleMyst**
Resolves #6, must be merged after #21

### CODE REVIEW

File: js/canvasmanager.ts [hunk 1]
```diff
@@ -178,6 +198,7 @@ export class CanvasManager extends StrictEventEmitter<CanvasManagerEvents>() {
       }).filter(obj => obj !== null).map(obj => obj!);
     }
     this.redraw();
+    this.emit("change", ChangeType.SetContents);
```
Comment (Akuli) [inline js/canvasmanager.ts:201]: This is the only place where a SetContents change is emitted. I think we should just get rid of SetContents changes for the canvas, and add a comment saying how the change event isn't emitted when the content is set.

File: js/canvasmanager.ts [hunk 2]
```diff
@@ -3,9 +3,10 @@ import { Pen } from "./drawobjects/pen";
 import { Circle } from "./drawobjects/circle";
 
 import { Point, LineMode, DrawObject } from "./drawobjects/drawobject";
+import { ChangeType } from "./editor";
```
Comment (Akuli) [inline js/canvasmanager.ts:6]: The canvas shouldn't depend on the editor. If the canvas manager really needs `ChangeType` (see my other review comment), then `ChangeType` should be moved to `utils.ts`.

**Review by Akuli — CHANGES_REQUESTED** [review]

**Review by Akuli — CHANGES_REQUESTED** [review]
Please merge dev branch. It's hard to review now with so much unrelated changes.

**Comment by PurpleMyst** [comment]
What do you mean? It seems to me they're at the same point? Or, at least, anytime I tell the git cli to merge them I get "already up to date"

**Review by Akuli — APPROVED** [review]

**Comment by PurpleMyst** [comment]
Nevermind, I did it. Man, my git is rusty.

**Comment by Akuli** [comment]
I recommend `git lola`, aka `git log --graph --decorate --pretty=oneline --abbrev-commit --all`

**Comment by PurpleMyst** [comment]
I have that (under a different name), I'd just forgotten to pull from origin for the dev branch

**Pull request merged**